src/approaches/interactive_learning_approach.py

```diff
@@ -0,0 +1,178 @@
+                 true_predicates: Set[Predicate],
+                 initial_options: Set[ParameterizedOption],
+                 types: Set[Type],
+                 action_space: Box,
+                 train_tasks: List[Task]) -> None:
+        # Only the teacher is allowed to know about the initial predicates
+        self._known_predicates = {p for p in true_predicates
+                                  if p.name in CFG.interactive_known_predicates}
+        predicates_to_learn = true_predicates - self._known_predicates
+        self._teacher = _Teacher(true_predicates, predicates_to_learn)
+        # All seen data
+        self.dataset: List[ActionTrajectory] = []
+        self.ground_atom_dataset: List[List[Set[GroundAtom]]] = []
+        # No cheating!
+        self._predicates = {_strip_predicate(p) for p in predicates_to_learn}
+        del true_predicates
+        del predicates_to_learn
+        super().__init__(simulator, self._predicates, initial_options,
+                         types, action_space, train_tasks)
+        self._operators: Set[Operator] = set()
+
+    @property
+    def is_learning_based(self) -> bool:
+        return True
+
+    def _get_current_operators(self) -> Set[Operator]:
+        assert self._operators, "Operators not learned"
+        return self._operators
+
+    def _get_current_predicates(self) -> Set[Predicate]:
+        return self._predicates
+
+    def learn_from_offline_dataset(self, dataset: Dataset) -> None:
+        # Get data from teacher
+        ground_atom_data = self._teacher.generate_data(dataset)
+        self.dataset.extend(dataset)
+        self.ground_atom_dataset.extend(ground_atom_data)
+        # Learn predicates
+        for pred in self._predicates:
+            print("\nPredicate:", pred)
+            assert pred not in self._known_predicates
+            positive_examples = []
+            negative_examples = []
+            # Positive examples
+            for i, trajectory in enumerate(self.ground_atom_dataset):
+                for j, ground_atom_set in enumerate(trajectory):
+                    state = self.dataset[i][0][j]
+                    positives = [state.vec(ground_atom.objects)
+                                 for ground_atom in ground_atom_set
+                                 if ground_atom.predicate == pred]
+                    positive_examples.extend(positives)
+            # Negative examples - assume unlabeled is negative for now
+            for (ss, _) in self.dataset:
+                for state in ss:
+                    possible = [state.vec(choice)
+                                for choice in get_object_combinations(
+                                                  list(state),
+                                                  pred.types,
+                                                  allow_duplicates=False)]
+                    negatives = []
+                    for (ex, pos) in itertools.product(possible,
```

Comment:
This is also something we can refactor later on if we're seeing slowness during classification dataset generation.